src/app/api/search/route.ts

```diff
@@ -0,0 +1,22 @@
+// search google books api for books matching search term, return results
+export async function GET(request: Request) {
+  const params = new URLSearchParams(request.url.split('?')[1]);
+  const term = params.get('q');
+  const startIndex = params.get('startIndex') || 0;
+  const maxResults = params.get('maxResults') || 10;
+  const orderBy = params.get('orderBy') || 'relevance';
+
+  let results;
+
+  try {
+    const response = await fetch(
+      `https://www.googleapis.com/books/v1/volumes?key=${process.env.GOOGLE_BOOKS_API_KEY}&q=${term}&startIndex=${startIndex}&maxResults=${maxResults}&orderBy=${orderBy}`
```

Comment:
https://developers.google.com/books/docs/v1/using#APIKey:~:text=To%20keep%20your,need%20any%20encoding.